project/Component/NefCompiler/Instances/NefCompilerSystem.swift

```diff
@@ -26,27 +27,32 @@ class NefCompilerSystem: CompilerSystem {
             content <- self.reorganizeHeaders(page: page),
             linkers <- self.dependencies(platform: platform),
             sources <- self.sources(inPlayground: playground),
-                    |<-self.compile(content: content.get, filename: filename, inPlayground: playground, andProject: project, sources: sources.get, platform: platform, frameworks: frameworks, linkers: linkers.get),
+                    |<-self.compile(content: content.get, filename: filename,
+                                    inPlayground: playground, atNefPlayground: nefPlayground,
+                                    sources: sources.get,
+                                    platform: platform,
+                                    frameworks: frameworks, linkers: linkers.get),
         yield: ())^
     }
     
     // MARK: - steps <shell>
-    private func createStructure(project: URL, cached: Bool) -> EnvIO<CompilerSystemEnvironment, CompilerSystemError, Void> {
+    private func createStructure(xcworkspace: URL, nefPlayground: NefPlaygroundURL, cached: Bool) -> EnvIO<CompilerSystemEnvironment, CompilerSystemError, Void> {
         EnvIO { env in
-            let cleanBuildIO = env.fileSystem.remove(itemPath: Path(project: project, action: .build).url.path).handleError { _ in }
-            let cleanLogIO = env.fileSystem.remove(itemPath: Path(project: project, action: .log).url.path).handleError { _ in }
-            let cleanRootIO = env.fileSystem.remove(itemPath: Path(project: project, action: .root).url.path).handleError { _ in }
+            let cleanBuildIO = env.fileSystem.remove(itemPath: nefPlayground.appending(.build).path).handleError { _ in }
```

Comment:
Maybe we can add an `ignoreError` extension function in Bow for `handleError { _ in }`.

Reply:
yep, I agree, I will add it  ^^